src/matter/interaction/SubscriptionHandler.ts

```diff
@@ -28,30 +29,31 @@ export class SubscriptionHandler {
         private readonly minIntervalFloorMs: number,
         private readonly maxIntervalCeilingMs: number,
     ) {
-        attributes.forEach(({ attribute }) => attribute.addListener(this.listener));
-        this.intervalId = setTimeout(() => this.sendUpdate(), this.minIntervalFloorMs);
+        attributes.forEach(({ attribute }) => attribute.addMatterListener(this.listener));
+        this.updateTimer = Time.get().getTimer(this.minIntervalFloorMs, () => this.sendUpdate()).start();
+        this.updateTimer.start();
     }
 
     async sendUpdate() {
         const now = Date.now();
         const timeSinceLastUpdateMs = now - this.lastUpdateTimeMs;
         if (timeSinceLastUpdateMs < this.minIntervalFloorMs) {
-            clearTimeout(this.intervalId);
-            this.intervalId = setTimeout(() => this.sendUpdate(), this.minIntervalFloorMs - timeSinceLastUpdateMs);
+            this.updateTimer.stop();
+            this.updateTimer = Time.get().getTimer(setTimeout(() => this.sendUpdate(), this.minIntervalFloorMs - timeSinceLastUpdateMs);
```

Comment:
should this not also be changed toTime.getTimer(...).start()... and also without a setTimeout in it?